Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift

```diff
@@ -57,29 +58,35 @@ struct MessagePayload<ExtraData: ExtraDataTypes>: Decodable {
     
     init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: MessagePayloadsCodingKeys.self)
-        id = try container.decode(String.self, forKey: .id)
-        type = try container.decode(MessageType.self, forKey: .type)
-        user = try container.decode(UserPayload<ExtraData.User>.self, forKey: .user)
-        createdAt = try container.decode(Date.self, forKey: .createdAt)
-        updatedAt = try container.decode(Date.self, forKey: .updatedAt)
-        deletedAt = try container.decodeIfPresent(Date.self, forKey: .deletedAt)
-        text = try container.decode(String.self, forKey: .text).trimmingCharacters(in: .whitespacesAndNewlines)
-        isSilent = try container.decodeIfPresent(Bool.self, forKey: .isSilent) ?? false
-        command = try container.decodeIfPresent(String.self, forKey: .command)
-        args = try container.decodeIfPresent(String.self, forKey: .args)
-        parentId = try container.decodeIfPresent(String.self, forKey: .parentId)
-        showReplyInChannel = try container.decodeIfPresent(Bool.self, forKey: .showReplyInChannel) ?? false
-        mentionedUsers = try container.decode([UserPayload<ExtraData.User>].self, forKey: .mentionedUsers)
-        // backend returns `thread_participants` only if message is a thread, we are fine with to have it on all messages
-        threadParticipants = try container.decodeIfPresent([UserPayload<ExtraData.User>].self, forKey: .threadParticipants) ?? []
-        replyCount = try container.decode(Int.self, forKey: .replyCount)
-        latestReactions = try container.decode([MessageReactionPayload<ExtraData>].self, forKey: .latestReactions)
-        ownReactions = try container.decode([MessageReactionPayload<ExtraData>].self, forKey: .ownReactions)
-        reactionScores = try container
-            .decodeIfPresent([String: Int].self, forKey: .reactionScores)?
-            .mapKeys { MessageReactionType(rawValue: $0) } ?? [:]
-        attachments = try container.decode([AttachmentPayload<ExtraData.Attachment>].self, forKey: .attachments)
-        extraData = try ExtraData.Message(from: decoder)
+        
+        if let message = try? container.decode(Self.self, forKey: .message) {
+            self = message
+        } else {
```

Comment:
+1 for creativity but please let's not do workarounds like this 😅 What about:
```
struct MessagePayloadBox(Crate/Wrapper/...??)<ExtraData: ExtraDataTypes> {
    let message: MessagePayload<ExtraData>
}
```

Reply:
Ok, I doubted this solution also :) 
`WrappedMessagePayload`?

Reply:
You could even do `MessagePayload.Wrapped` 🤯 Next time :)